pydantic/main.py

```diff
@@ -887,7 +887,7 @@ class MyModel(BaseModel, extra='allow'):
 
     def __iter__(self) -> TupleGenerator:
         """So `dict(model)` works."""
-        yield from self.__dict__.items()
+        yield from [(k, v) for (k, v) in self.__dict__.items() if not k.startswith('_')]
```

Comment:
This seems like it should be it's own PR. The repr thing is one thing, the __iter__ change another